lib/screens/input_password.dart

```diff
@@ -1,13 +1,93 @@
 import 'package:flutter/material.dart';
+import 'package:niira/models/game.dart';
+import 'package:niira/models/player.dart';
+import 'package:niira/screens/waiting_for_game_to_start.dart';
+import 'package:niira/services/auth/auth_service.dart';
+import 'package:niira/services/database/database_service.dart';
+import 'package:provider/provider.dart';
+
+class InputPasswordScreen extends StatefulWidget {
+  final Game game;
+  InputPasswordScreen({@required this.game});
+  @override
+  _InputPasswordScreenState createState() => _InputPasswordScreenState();
+}
+
+class _InputPasswordScreenState extends State<InputPasswordScreen> {
+  final _formKey = GlobalKey<FormState>();
 
-class InputPasswordScreen extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       key: Key('inputPasswordScreen'),
+      appBar: AppBar(
+        automaticallyImplyLeading: true,
+        title: Text('Password'),
+      ),
+      floatingActionButton: FloatingActionButton.extended(
+        key: Key('input_password_screen_submit_btn'),
+        label: Text(
+          'Next',
+          style: TextStyle(color: Colors.white),
+        ),
+        icon: Icon(
+          Icons.navigate_next,
+          color: Colors.white,
+        ),
+        onPressed: () async {
+          if (_formKey.currentState.validate()) {
+            // get player details
+            final userId = await context.read<AuthService>().currentUserId;
+            final username =
+                await context.read<DatabaseService>().getUserName(userId);
+
+            // add user as player to game
+            final player = Player(
+              id: userId,
+              username: username,
+              isTagger: false,
+              hasBeenTagged: false,
+              hasItem: false,
+              isAdmin: false,
+            );
+            await context
+                .read<DatabaseService>()
+                .joinGame(widget.game.id, player);
+
+            // navigate to waiting screen
+            await Navigator.push<dynamic>(
+              context,
+              MaterialPageRoute<dynamic>(
+                builder: (context) => WaitingForGameToStartScreen(),
+              ),
+            );
+          }
+        },
+      ),
       body: Container(
         child: Center(
-          child: Text('input password screen'),
+          child: Form(
```

Comment:
Long build functions are definitely more difficult to review than splitting up into smaller widgets.  I've kind of lost what the code is doing at this point.

Reply:
OK I think I get it, this is just the body of the Scaffold still and it just has the one thing - a password field that... hmmm, some commenting would really help here mate :-)

Reply:
I think I'll just approve and you can either push it through or come back and make it easier to review, up to you.  I'm pretty tired.  Comments and splitting up into smaller widgets both make it easier to read and understand someone else's code.

Reply:
I apologise again for the tone here mate - I was quite sleep deprived and having a bad day, I should have left the review for when I was in a better head space. I'll do better in future 👍

Reply:
allgood man! hope you had a better day today!